### PR TITLE
182 Develop utilities for NetCDF metadata extraction

### PR DESCRIPTION
**Pull request by david-inglish**
## Description
- Implements get_rtc_s1_product_metadata() to retrieve RTC-S1 HDF5 product file metadata for use in ISO files.

## Affected Issues
- #182
- 
## Testing
- Added test_get_rtc_s1_product_metadata() to test_metadata_utils.py to validate h5py library usage.

### CODE REVIEW

**Comment by collinss-jpl** [comment]
Hi @david-inglish, I'm still reviewing this branch, but I noticed that the unit tests are failing in Jenkins since h5py needs to be added to requirements.txt so it gets installed within the container.

**Review by collinss-jpl — COMMENTED** [review]
Approach looks solid, just needs a few updates and this should be good to merge. Thanks David!

File: src/opera/util/metadata_utils.py [hunk 1]
```diff
@@ -187,3 +192,96 @@ def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name):
...
+        data is copied from the h5py group object to a python dict.
+    """
+    converted_dict = {}
+    for key,val in group_object.items():
+
+        if isinstance(val, h5py._hl.dataset.Dataset):
+            if type(val[()]) is np.ndarray:
+                if isinstance(val[0], bytes) or isinstance(val[0], np.bytes_):
+                    # decode bytes to str
+                    converted_dict[key] = val[()].astype(str)
```
[the remark this answers is not in the document]
Comment (david-inglish) [reply src/opera/util/metadata_utils.py]: Yes, the resulting dtype in the arrays is "object" when asstr is used.  It was a strange value when astype(str) is used.  Also, I was able to use asstr for non-array Datasets so the code is consistent.

File: src/opera/util/metadata_utils.py [hunk 2]
```diff
@@ -187,3 +192,96 @@ def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name):
...
+        group path within the HDF5 file.
+
+    Returns
+    -------
+    group_dict : dict
+        python dict containing variable data from the group path location.
+    """
+    group_dict = {}
+    with h5py.File(file_name, 'r') as hf:
+        group_object = hf.get(group_path)
```
[the remark this answers is not in the document]
Comment (david-inglish) [reply src/opera/util/metadata_utils.py:214]: It looks like get() just returns None if the group is not located.

**Review by collinss-jpl — APPROVED** [review]
Looks like CI is now passing, I know library issues can be nightmare, thanks for seeing this one through!

**Pull request merged**